### ISSUE TITLE
Saving an order causes a tax breakdown error

### ISSUE THREAD

**Issue by ryanmitchell** [lunarphp/lunar#882]
Using rc-0.2

We have a process where after creating an order we sync with a 3rd party warehousing service and add a id to the `meta` so the two are kept in sync:

```php
            $meta = $order->meta ?? (object)[];
            $meta->pearl_id = 'xxxx';
            $order->meta = $meta;
            $order->save();
```

However now that we're using the Lunar tax service, we're getting an error at that point:

```
Lunar\Exceptions\InvalidDataTypeValueException
Value was "object" expected "int"
```

coming from `vendor/lunarphp/lunar/packages/core/src/Base/Casts/TaxBreakdown.php:27`

It seems the tax breakdown is being saved to the database as:

```json
[{"total": {"value": 0, "unitQty": 1, "currency": {"id": 1, "code": "GBP", "name": "British Pound", "default": 1, "enabled": 1, "created_at": "2023-01-18T08:26:30.000000Z", "updated_at": "2023-01-19T07:18:34.000000Z", "exchange_rate": "1.0000", "decimal_places": 2}}, "identifier": "T0", "percentage": 0, "description": "UK"}, {"total": {"value": 55, "unitQty": 1, "currency": {"id": 1, "code": "GBP", "name": "British Pound", "default": 1, "enabled": 1, "created_at": "2023-01-18T08:26:30.000000Z", "updated_at": "2023-01-19T07:18:34.000000Z", "exchange_rate": "1.0000", "decimal_places": 2}}, "identifier": "T20", "percentage": 20, "description": "UK"}]
```

instead of 

```json
[{"total": 0, "identifier": "T0", "percentage": 0, "description": "UK"}, {"total": 55, "identifier": "T20", "percentage": 20, "description": "UK"}]
```

Maybe something to do with the casting? Is there anything we can do about this?

**Comment by glennjacobs** [comment]
@alecritson could you take a look at this? Would be good to get a fix into 0.2 before it's released.

**Comment by ryanmitchell** [comment]
Closed by #887
